R2Northstar/NorthstarLauncher
# Rework MAD `cleanup` handles

### Changes

* Mod is not marked as successfully installed when the process failed;
* When mod downloading/extraction fails, mod directory is removed.

### Testing

1. Install this PR;
2. Enable MAD (`allow_mod_auto_download 1`);
3. Try to join Space battle server and cancel mod downloading/extraction;
4. Check your local profile, the remote mod should not appear in the `runtime/remote/mods` directory;
5. Check your temporary directory (for me on Windows, it's `C:\Users\Remy\AppData\Local\Temp`), there should be no mod archive.

---

Closes #756.


---

##### TODOs

- [x] ZIP archive is not deleted when download is cancelled (but it is when cancelling extraction)
- [x] Mod is not removed when cancelling archive extraction
